tests/ui_tools/test_messages.py

```diff
@@ -1585,14 +1587,120 @@ def test_keypress_EDIT_MESSAGE(
             # fmt: on
         ],
     )
-    def test_transform_content(self, mocker, raw_html, expected_content):
+    def test_transform_content(self, raw_html: str, expected_content: Any) -> None:
         expected_content = expected_content.replace("{}", QUOTED_TEXT_MARKER)
 
         content, *_ = MessageBox.transform_content(raw_html, SERVER_URL)
 
         rendered_text = Text(content)
         assert rendered_text.text == expected_content
 
+    @pytest.mark.parametrize(
+        "raw_html, expected_message_links",
+        [
+            (
+                """
+                <p><a href="https://github.com/zulip/zulip-terminal/pull/1">https://github.com/zulip/zulip-terminal/pull/1</a></p>
+                """,
+                {
+                    "https://github.com/zulip/zulip-terminal/pull/1": (
+                        "github.com",
+                        1,
+                        True,
+                    )
+                },
+            ),
+            (
+                """
+                <p><a href="https://foo.com">https://foo.com</a></p>
+                """,
+                {"https://foo.com": ("https://foo.com", 1, False)},
+            ),
+            (
+                """
+                <p><a href="#narrow/stream/206-zulip-terminal/topic/announce">https://chat.zulip.zulip/#narrow/stream/206-zulip-terminal/topic/announce</a></p>
+                """,
+                {
+                    "https://chat.zulip.zulip#narrow/stream/206-zulip-terminal/topic/announce": (  # noqa: E501
+                        "https://chat.zulip.zulip/#narrow/stream/206-zulip-terminal/topic/announce",
+                        1,
+                        True,
+                    )
+                },
+            ),
+            (
+                """
+                <p><a href="https://github.com/zulip/zulip-terminal">test</a></p>
+                """,
+                {"https://github.com/zulip/zulip-terminal": ("test", 1, True)},
+            ),
+        ],
+        ids=[
+            "github_pr",
+            "external_link_no_footlink",
+            "internal_link",
+            "github_link_with_text",
+        ],
```

Comment:
Let's have these ids inline.

Also the first is a github_pr (link) but it'd be useful to clarify what form it takes, ie. the link matches the text in that first case.

Reply:
`github_pr_link_matches_text`?